panedr/tests/test_edr.py

```diff
@@ -86,79 +112,42 @@ def test_columns(self):
             print('These columns differ from the reference (displayed as read):')
             print(columns[ref_columns != columns])
             print('The corresponding names displayed as reference:')
             print(ref_columns[ref_columns != columns])
-        self.assertTrue(ref_columns.shape == columns.shape,
-                        'The number of column read is unexpected.')
-        self.assertTrue(numpy.all(ref_columns == columns),
-                        'At least one column name was misread.')
+        assert ref_columns.shape == columns.shape, \
+               'The number of columns read is unexpected.'
+        assert numpy.all(ref_columns == columns), \
+               'At least one column name was misread.'
 
-    def test_times(self):
+    def test_times(self, edr):
         """
         Test that the time is read correctly when dt is regular.
         """
-        df = panedr.edr_to_df(EDR)
-        xvg = read_xvg(EDR_XVG)
-        ref_time = xvg[:, 0]
-        time = df[u'Time'].as_matrix()
-        self.assertTrue(numpy.allclose(ref_time, time, atol=5e-7))
-
-    def test_times_double(self):
-        """
-        Test that the time is read correctly when dt is regular.
-        """
-        df = panedr.edr_to_df(EDR_DOUBLE)
-        xvg = read_xvg(EDR_DOUBLE_XVG)
-        ref_time = xvg[:, 0]
-        time = df[u'Time'].as_matrix()
-        self.assertTrue(numpy.allclose(ref_time, time, atol=5e-7))
-
-    def test_times_irregular(self):
-        """
-        Test that the time is read correctly when dt has irregularities.
-        """
-        df = panedr.edr_to_df(EDR_IRREGULAR)
-        xvg = read_xvg(EDR_IRREGULAR_XVG)
-        ref_time = xvg[:, 0]
-        time = df[u'Time'].as_matrix()
-        self.assertTrue(numpy.allclose(ref_time, time, atol=5e-7))
+        time = edr.df[u'Time'].as_matrix()
+        assert numpy.allclose(edr.xvgtime, time, atol=5e-7)
 
-    def test_content(self):
+    def test_content(self, edr):
         """
         Test that the content of the DataFrame is the expected one.
         """
-        df = panedr.edr_to_df(EDR)
-        xvg = read_xvg(EDR_XVG)
-        ref_content = xvg[:, 1:]  # The time column is tested separately
-        content = df.iloc[:, 1:].as_matrix()
-        print(ref_content - content)
-        self.assertTrue(numpy.allclose(ref_content, content, atol=5e-7))
-
-    def test_content_double(self):
-        """
-        Test that the content of the DataFrame is the expected one.
-        """
-        df = panedr.edr_to_df(EDR_DOUBLE)
-        xvg = read_xvg(EDR_DOUBLE_XVG)
-        ref_content = xvg[:, 1:]  # The time column is tested separately
-        content = df.iloc[:, 1:].as_matrix()
-        print(ref_content - content)
-        self.assertTrue(numpy.allclose(ref_content, content, atol=1e-12))
+        content = edr.df.iloc[:, 1:].as_matrix()
+        print(edr.xvgdata - content)
+        assert numpy.allclose(edr.xvgdata, content, atol=edr.xvgprec/2)
```

Comment:
`edr.xvgprec` can be 0.1, 0.01, 0.001, etc. matching the precision in the xvg. If we want to compare to the xvg text data we can't allow a tolerance equal to the precision. That'd mean that the number 1.234 (which has precision 0.001) would be matched to 1.235 or 1.233. Using half the precision as the tolerance ensures that only numbers with the same text representation match.